c++/src/capnp/rpc-twoparty-test.c++

```diff
@@ -879,6 +879,271 @@ KJ_TEST("Streaming over RPC then unwrap with CapabilitySet") {
+  test::TestRealtimeStreaming::Client serverCap(kj::mv(ownServer));
+
+  TwoPartyClient tpClient(*pipe.ends[0]);
+  TwoPartyClient tpServer(*pipe.ends[1], serverCap, rpc::twoparty::Side::SERVER);
+
+  auto cap = tpClient.bootstrap().castAs<test::TestRealtimeStreaming>();
+
+  // Try to stream a capability and check that it throws
+  auto req = cap.doFailRealtimeStreamRequest();
+  req.setCap(cap);
+
+  kj::Maybe<kj::Exception> maybeException = kj::runCatchingExceptions([&]() {
+    auto ignored = req.send();
+  });
+
+  KJ_IF_SOME(e, maybeException) {
+    KJ_EXPECT(e.getType() == kj::Exception::Type::FAILED);
+  } else {
+    KJ_FAIL_EXPECT("should have thrown");
+  }
+}
+
+KJ_TEST("Realtime streaming bypasses streaming flow control") {
+  // This tests that realtime streaming is NOT blocked by streaming call backpressure.
+  // Realtime messages should go through even when streaming calls are backed up.
+
+  kj::EventLoop loop;
+  kj::WaitScope waitScope(loop);
+
+  auto pipe = kj::newTwoWayPipe();
+
+  size_t window = 1024;
+  size_t clientWritten = 0;
+  size_t serverWritten = 0;
+
+  pipe.ends[0] = kj::heap<MockSndbufStream>(kj::mv(pipe.ends[0]), window, clientWritten);
+  pipe.ends[1] = kj::heap<MockSndbufStream>(kj::mv(pipe.ends[1]), window, serverWritten);
+
+  auto ownServer = kj::heap<TestRealtimeStreamingImpl>();
+  auto& server = *ownServer;
+  test::TestRealtimeStreaming::Client serverCap(kj::mv(ownServer));
+
+  TwoPartyClient tpClient(*pipe.ends[0]);
+  TwoPartyClient tpServer(*pipe.ends[1], serverCap, rpc::twoparty::Side::SERVER);
+
+  auto cap = tpClient.bootstrap().castAs<test::TestRealtimeStreaming>();
+
+  // Send normal streaming requests until flow control kicks in
+  kj::Promise<void> promise = kj::READY_NOW;
+  uint count = 0;
+  while (promise.poll(waitScope)) {
+    promise.wait(waitScope);
+
+    auto req = cap.doStreamRequest();
+    req.setI(++count);
+    promise = req.send();
+  }
+
+  // Streaming is now blocked. Send a realtime request - it should NOT be blocked.
+  auto req = cap.doRealtimeStreamRequest();
+  req.setJ(100);
+  auto realtimePromise = req.send();
+  KJ_EXPECT(realtimePromise.poll(waitScope), "Realtime message should bypass streaming backpressure");
+  realtimePromise.wait(waitScope);
+
+  // Send another realtime request to confirm
+  req = cap.doRealtimeStreamRequest();
+  req.setJ(3);
+  realtimePromise = req.send();
+  KJ_EXPECT(realtimePromise.poll(waitScope), "Realtime message should bypass streaming backpressure");
+  realtimePromise.wait(waitScope);
+
+  // Fulfill the remaining streaming requests
+  server.setAutoFulfill(true);
+
+  // Finish and verify both realtime messages got through
+  auto finishReq = cap.finishStreamRequest();
+  auto result = finishReq.send().wait(waitScope);
+
+  // Both realtime messages should have been delivered despite streaming backpressure
+  KJ_EXPECT(result.getTotalJ() == 103);
+}
+
+KJ_TEST("Realtime streaming drops messages when network queue is full") {
```

Comment:
I implemented your suggestion to drop realtime messages when the network is congested, and I added this test to verify it.